.pre-commit-config.yaml

```
@@ -1266,6 +1267,14 @@ repos:
         files:
           ^airflow-ctl/src/airflowctl/api/operations.py|airflow-ctl/src/airflowctl/cli/.*\.py$
         additional_dependencies: ['rich>=12.4.4']
+      - id: mockery
+        name: Generate mocks for go
+        entry: --entrypoint ./scripts/in_container/run_mockery.sh vektra/mockery:3
```

```suggestion
        entry: --workdir /src/go-sdk vektra/mockery
```

It turns out you can put any `docker run` cli args here.

---

Yup, I had looked into that as well, but I thought it'd not work because they explicitly set the workdir to pwd [here](https://github.com/pre-commit/pre-commit/blob/main/pre_commit/languages/docker.py#L151). I wasn't sure how docker responds to multiple workdir params. Will try it out though, if it works that saves a lot of effort.

---

Did you try doing something like:
```
- id: go-mockery
  name: Generate mocks for Go
  language: docker_image
  entry: sh -c "cd go-sdk && mockery"
  types: [go]
  files: ^go-sdk/
  exclude: mocks/.*\.go$
  pass_filenames: false
  additional_dependencies: [vektra/mockery:3]
```

---

Another qn is, do we not need to supply the mockery.yml to the command? (not sure)

---

We don't need to supply the config file if we execute it from the right dir -- it'll look at `.mockery.yaml` in the CWD by default

---

For the entry value we need something that'll get fit into a docker run command, so that won't work. But I think we can use @ashb's suggestion of adding a workdir would work, since this gets appended at the end of the command and from my experimenting docker uses the latest value of workdir it multiple are provided so we can get around their hardcoded workdir.

---

done